FLT/Deformations/Lemmas.lean

```diff
@@ -0,0 +1,168 @@
+import Mathlib.RingTheory.LocalRing.ResidueField.Basic
+import Mathlib.Topology.Algebra.Algebra.Equiv
+import Mathlib.Topology.Algebra.LinearTopology
+import Mathlib.Topology.Instances.Matrix
+import Mathlib.Topology.UniformSpace.DiscreteUniformity
+
+lemma exists_ideal_isMaximal_and_isOpen_of_isLinearTopology
+    (R : Type*) [CommRing R] [TopologicalSpace R] [IsTopologicalRing R]
+    [IsLinearTopology R R] [Nontrivial R] [T0Space R] :
+    ∃ I : Ideal R, I.IsMaximal ∧ IsOpen (X := R) I := by
+  obtain ⟨I, hI, hI'⟩ := IsLinearTopology.hasBasis_open_ideal.mem_iff.mp
+    ((isClosed_singleton (x := (1 : R))).isOpen_compl.mem_nhds (x := 0) (by simp))
+  obtain ⟨J, hJ, hIJ⟩ := Ideal.exists_le_maximal I (by simpa [Ideal.eq_top_iff_one] using hI')
+  exact ⟨J, hJ, AddSubgroup.isOpen_mono (H₁ := I.toAddSubgroup) (H₂ := J.toAddSubgroup) hIJ hI⟩
+
+/-- The continuous group homomorphism on units induced by a `ContinuousMonoidHom`. -/
+@[simps!]
+def Units.mapₜ {M N : Type*} [Monoid M] [Monoid N] [TopologicalSpace M] [TopologicalSpace N]
+    (f : M →ₜ* N) : Mˣ →ₜ* Nˣ :=
```

Comment:
Almost everything in this file looks suitable for mathlib. Should they be moved to the Mathlib folder?